File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.py
import heapq
def solution(scoville, K):
    answer = 0
    heap = []
    for i in scoville:
        heapq.heappush(heap,i)
    
    while(heap[0] < K and len(heap) >= 2):
        answer += 1
        a = heapq.heappop(heap)
        b = heapq.heappop(heap)
        heapq.heappush(heap, a+(b*2))
    if(heap[0] < K):
        answer = -1
    return answer

File: 프로그래머스/lv2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        PriorityQueue<Integer> q = new PriorityQueue();
        for(int i : scoville){
            q.add(i);
        }
        
        while(q.peek() < K && q.size() >= 2){
            answer++;
            int a = q.poll();
            int b = q.poll();
            int ans = a+(b*2);
            q.add(ans);
        }
        
        if(q.peek() < K){
            answer= -1;
        }
        
        return answer;
    }
}